docs/kepler_model_server/pipeline.md

```diff
@@ -1,29 +1,20 @@
 # Training Pipeline
 
-As shown in the below figure, a power model can be attributed by the path it was trained. Training pipeline is an abstract of power model training that applies a set of learning methods to a different combination of energy source labels, available metrics, and idle/control plane power isolation choices to each specific group of nodes/machines. 
+Model server serves extensive choices of power models. Training pipeline is an abstract of power model training that applies a set of learning methods to a different combination of energy sources, power isolation methods, available energy-related metrics.  
 
-A training pipeline starts from reading the Kepler-exporting metrics from Prometheus query (prom) and finally submits an archived models to the model database (model DB). 
+## Pipeline
+`pipeline` is composed of three steps, `extract`, `isolate`, and `train`, as shown below. Kepler exports energy-related metrics to the Prometheus in accumulated number over time. `extract` is to convert the aggregated metrics to per-second values and clean up the data separately for each `feature group`. `isolate` is to compute an idle power or control plane power and remove it from the power label of each `energy source`. `train` is to apply each `trainer` to create multiple choices of power models based on the preprocessed data. 
 
-From Kepler queries, the default extractor generates a dataframe with the following columns.
+![](../fig/pipeline_plot.png)
 
-timestamp|features|labels[unit]\_[#unit]\_[component]\_power|node type
----|---|---|---
-timestamp|e.g.,<br>cgroupfs_cpu_usage_us<br>cgroupfs_memory_usage_bytes<br>cgroupfs_system_cpu_usage_us<br>cgroupfs_user_cpu_usage_us|e.g.,<br>package_0_package_power<br>package_1_package_power<br>package_0_core_power<br>package_1_core_power<br>package_0_uncore_power<br>package_1_uncore_power<br>package_0_dram_power<br>package_1_dram_power|node_type
+- Learn more about `energy source` from [Feature groups](#feature-groups) section.
+- Learn more about `feature group` from [Energy source](#labeling-energy-source) section.
+- Learn more about `isolate`, `AbsPower`, and `DynPower` from [Power isolation](#power-isolation) section.
```

Comment:
Learn more about `isolate`, `AbsPower`, and `DynPower` from
->
Learn more about the`isolate` step, the resulted `AbsPower`, and `DynPower` power models from